asf_public_discourse_home_decarbonisation/analysis/buildhub/extract_faqs.py

```diff
@@ -0,0 +1,245 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def create_argparser() -> argparse.ArgumentParser:
+    """
+    Creates an argument parser that can receive the following arguments:
+    - category: category or sub-forum (defaults to "119_air_source_heat_pumps_ashp")
+    - collection_date_time: collection date/time (defaults to "231120")
+    Returns:
+        argparse.ArgumentParser: argument parser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--forum",
+        help="Forum that we are looking at",
+        default="bh",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--category",
+        help="Category or sub-forum",
+        default="119_air_source_heat_pumps_ashp",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--collection_date_time",
+        help="Collection date/time",
+        default="231120",
+        type=str,
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def extract_questions(
+    sentences: List[str], num_of_words: int = 5, min_length: int = 5
+) -> List[str]:
+    """
+    Extracts and returns a curated list of questions from a list of sentences, specifically filtering out certain types of questions. This function is designed to identify and include questions that are more substantial and exclude shorter, less informative questions such as "any thoughts?" or "why?".
+
+    The function utilises a combination of punctuation analysis (identifying sentences ending with '?') and the presence of certain question words at the beginning. However, it also employs a set of criteria to exclude questions that are deemed too brief or lacking in content, even if they technically qualify as questions.
+
+    Args:
+        sentences (List[str]): A list of sentences from which questions are to be extracted.
+        num_of_words (int, optional): The maximum number of words allowed for a question to be considered valid. Defaults to 5.
+        min_length (int, optional): The minimum length of a question to be considered valid. Defaults to 5.
+
+    Returns:
+        List[str]: A list of strings, each a question identified from the input sentences. Questions are filtered to exclude those that are brief or deemed less informative.
+
+    Note:
+        - The function uses predefined question words to identify potential questions but excludes questions that are brief or deemed less informative, such as "why?" or "any thoughts?".
+        - This exclusion is based on the specific requirements for the type of questions the function is intended to extract, focusing on more substantial and content-rich questions.
+    """
+    question_words = [
+        "what",
+        "how",
+        "where",
+        "when",
+        "who",
+        "why",
+        "which",
+        "really",
+        "any",
+        "what's",
+        "how's",
+        "really",
+        "thoughts",
+        "is",
+    ]
```

Comment:
Because they were coming up over and over again when looking at the duplicate questions

Reply:
Just thinking about this a little bit more since our stand-up, with the condition `if len(words) <= num_of_words and any(word.lower() in question_words for word in words):`, we are likely going to get rid of short annoying questions like "thoughts?" which don't tell us anything. I think this is part of the reason I included these here even if it is a bit counter intuitive. The `min_length` parameter then gets rid of other short words which aren't very informative but enables to have the flexibility to have acronyms such as 'MVHR?'.

Reply:
This was the reasoning at least, think it might be just useful to simplify the methodology a bit because it's becoming a bit convoluted in it's reasoning based on a few anecdotal data points.